sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultAdministrationUtil.java

```diff
@@ -0,0 +1,383 @@
+                "'roleScope'"));
+        Objects.requireNonNull(roleAssignmentName,
+            String.format(KeyVaultErrorCodeStrings.getErrorString(KeyVaultErrorCodeStrings.PARAMETER_REQUIRED),
+                "'roleAssignmentName'"));
+    }
+
+    static void validateRoleDefinitionParameters(KeyVaultRoleScope roleScope, String roleDefinitionName) {
+        Objects.requireNonNull(roleScope,
+            String.format(KeyVaultErrorCodeStrings.getErrorString(KeyVaultErrorCodeStrings.PARAMETER_REQUIRED),
+                "'roleScope'"));
+        Objects.requireNonNull(roleDefinitionName,
+            String.format(KeyVaultErrorCodeStrings.getErrorString(KeyVaultErrorCodeStrings.PARAMETER_REQUIRED),
+                "'roleDefinitionName'"));
+    }
+
+    @SuppressWarnings("BoundedWildcard")
+    static PagedResponse<KeyVaultRoleDefinition> transformRoleDefinitionsPagedResponse(
+        PagedResponse<RoleDefinition> pagedResponse) {
+
+        List<KeyVaultRoleDefinition> keyVaultRoleDefinitions = new ArrayList<>();
+
+        for (RoleDefinition roleDefinition : pagedResponse.getValue()) {
+            keyVaultRoleDefinitions.add(roleDefinitionToKeyVaultRoleDefinition(roleDefinition));
+        }
+
+        return new TransformedPagedResponse<>(keyVaultRoleDefinitions, pagedResponse);
+    }
+
+    static Response<KeyVaultRoleDefinition> transformRoleDefinitionResponse(Response<RoleDefinition> response) {
+        KeyVaultRoleDefinition keyVaultRoleDefinition = roleDefinitionToKeyVaultRoleDefinition(response.getValue());
+
+        return new TransformedResponse<>(keyVaultRoleDefinition, response);
+    }
+
+    static KeyVaultRoleDefinition roleDefinitionToKeyVaultRoleDefinition(RoleDefinition roleDefinition) {
+        List<KeyVaultPermission> keyVaultPermissions = new ArrayList<>();
+
+        for (Permission permission : roleDefinition.getPermissions()) {
+            keyVaultPermissions.add(
+                new KeyVaultPermission(permission.getActions(), permission.getNotActions(),
+                    permission.getDataActions().stream()
+                        .map(dataAction -> KeyVaultDataAction.fromString(dataAction.toString()))
+                        .collect(Collectors.toList()),
+                    permission.getNotDataActions().stream()
+                        .map(notDataAction -> KeyVaultDataAction.fromString(notDataAction.toString()))
+                        .collect(Collectors.toList())));
+        }
+
+        return new KeyVaultRoleDefinition(roleDefinition.getId(), roleDefinition.getName(),
+            KeyVaultRoleDefinitionType.fromString(roleDefinition.getType().toString()), roleDefinition.getRoleName(),
+            roleDefinition.getDescription(), KeyVaultRoleType.fromString(roleDefinition.getRoleType().toString()),
+            keyVaultPermissions, roleDefinition.getAssignableScopes().stream()
+            .map(roleScope -> KeyVaultRoleScope.fromString(roleScope.toString()))
```

Comment:
Not related to this PR, but let's begin a list of changes that should be made with Autorest transforms, and this can be the first.

`RoleScope` and `KeyVaultRoleScope` are the same class, where `RoleScope` is code generated and `KeyVaultRoleScope` is written by hand. We can use a Swagger transform rename on `RoleScope` to make it `KeyVaultRoleScope` and remove the need for this transformation.